src/smt/optimization_solver.h

```diff
@@ -33,63 +33,70 @@ namespace smt {
 /**
  * The optimization result of an optimization objective
  * containing:
- * - whether it's optimal or not
- * - if so, the optimal value, otherwise the value might be empty node or
- *   something suboptimal
+ * - the optimization result: SAT/UNSAT/UNKNOWN
+ * - the optimal value if SAT and bounded,
+ *   or an empty node if SAT and unbounded,
+ *   otherwise the value might be empty node or something suboptimal
+ * - whether the objective is unbounded
```

Comment:
Please move this to where the fields are documented. Most of it is already there, except for the corner cases.

Reply:
This is exactly the description for class OptimizationResult. So I think it's the appropriate location.